### ISSUE TITLE
Can't set independent layer legend in faceted layered chart

### ISSUE THREAD

**Issue by dechamps** [vega/vega-lite#6261]
[Consider](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEBuDGD2A7WBDALpAXGA2qCEkAZirAKYbbCRoCeADmVlIvALYCWiKANpADRQiHMjwAmzYpAC+-cAUgBnRrGb4ChHilpkATszzyNhNil0BrNVDqNJPLkzCyjxyGWTwxXAOZqXxqAQeeH0qfwDrBiZsSFZObj45CNdhUQkY1XCNaSyncNkwdQDIUwsrGijJWA5dWB5HZ2S3Dy9EXzDkwiCQv06FG2iWdi5eAVyNYhFxKshxvOScgOkAXRdFp2WkqDF0FDUYXgBXMkUDaiJJAEYBQMkAQRlBc8kAJhvIVRiAIRllnJygA):

```
{
  "vconcat": [{
    "facet": {"type": "nominal", "field": "f"},
    "spec": {
      "layer": [{
        "mark": { "type": "line" },
        "encoding": {
          "color": {
            "type": "nominal",
            "field": "c"
          }
        }
      }, {
        "mark": { "type": "circle" },
        "encoding": {
          "color": {
            "type": "nominal",
            "field": "c"
          }
        }
      }]
    }
  }],
  "data": {"values": [{"f": "1", "c": "A"}, {"f": "2", "c": "B"}]}
}
```
![visualization(53)](https://user-images.githubusercontent.com/866859/78452161-8c01ac00-7681-11ea-837b-b8e3627b9535.png)

I want to set `"resolve": {"legend": {"color": "independent"}}` to get independent legends but *only between the layers, not between the facets*.

If I [add the property under each layer](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEBuDGD2A7WBDALpAXGA2qCEkAZirAKYbbCRoCeADmVlIvALYCWiKANpADRQiHMjwAmzYpAC+-cAUgBnRrGb4ChHilpkATszzyNhNil0BrNVDqNJPLkzCyjxyGWTwxXAOZqXxqAQeeH0qfwDrBiZsSFZObj45CNdhUQkY1XCNaSynJIjIXTJFeB5oaLBqHjJvd3TKyCCQyS4xMkZENsQMaRyA2UrcyFMLKxooyVgOXVhqyDyh9wQvRF8w5MIm0MGNhRsK2PYuXgFcjWIRcUnIM6dc52TC4tLytUhq2s63rZbO9rr3D0+sZpABdFzAsH5SBidAoN7QXgAV2KBmoREkAEYBIFJABBGSCdGSABMOMakgAQjJQTkckA), it doesn't do anything.

If I [add the property under `spec`](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEBuDGD2A7WBDALpAXGA2qCEkAZirAKYbbCRoCeADmVlIvALYCWiKANpADRQiHMjwAmzYpAC+-cAUgBnRrGb4ChHilpkATszzyNhNil0BrNVDqNJPLkzCyjxyGWTwxXAOZqXxqAQeeH0qfwDrBiZsSFZObj45CNdhUQkY1XCNaSyncNkwdQDIUwsrGijJWA5dWB5HZ2S3Dy9EXzDkwiCQv06FG2iWdi5eAVyNYhFxKshxvOScgOkAXSTXXTJFeB5oQep673d0wshu0KguMTJGRCvEDGlFgieVtcgxdBQ1GF4AV02DNQiJIAIwCQKSACCMkEQMkACZwadJAAhGTLHI5IA), it makes legends independent between layers *and* facets, and I end up with 4 legends, which is not what I want:
![visualization(55)](https://user-images.githubusercontent.com/866859/78452266-40033700-7682-11ea-92c9-dd7bc56f6d06.png)
(trying to counteract that by adding `shared` at the upper levels [does nothing](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEBuDGD2A7WBDALpAXGA2qCEkAZirAKYbbCRoCeADmVlIvALYCWiKANpADRQiHMjwAmzYpAC+-cAUgBnRrGb4ChHilpkATszzyNhNil0BrNVDqNJPLkzCyjxyGWTwxXAOZqXxqAQeeH0qfwDrBiZsSFZObj45CNdhUQkY1XCNaSyncNkwdQDIUwsrGijJWA5dWB5HZ2S3Dy9EXzDkwiCQv06FG2iWdi5eAVyNYhFxKshxvOScgOkAXSTXXTJFeB5oQep673d0wshu0KguMTJGRCvEDGlFgkaFDa2dvcgDo7VT7Z6YooABZmMgSR7yFZrSBidAoX7QXgAV02BmoREkAEYBIFJABBGSCdGSABMONOkgAQjJli9IG9trtft9br8zpJgaDwTkckA)).

If I [add the property under `vconcat`](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEBuDGD2A7WBDALpAXGA2qCEkAZirAKYbbCRoCeADmVlIvALYCWiKANpADRQiHMjwAmzYpAC+-cAUgBnRrGb4ChHilpkATszzyNhNil0BrNVDqNJPLkzCyjxyGWTwxXAOZqXxqAQeeH0qfwDrBiZsSFZObj45CNdhUQkY1XCNaSyncNkwdQDIUwsrGijJWA5dWB5HZ2S3Dy9EXzDkwiCQv06FG2iWdi5eAVyNYhFxKshxvOScgOkAXRdGhV0yRXgeaEHqeu93dMLIbtCoLjEyRkRrxAxpRadlpKgxdBQ1GF4AVy2DNQiJIAIwCQKSACCMkEQMkACZwWdJAAhGTLHI5IA), I get independent legends between facets, but not between layers, the opposite of what I want:
![visualization(54)](https://user-images.githubusercontent.com/866859/78452193-c8350c80-7681-11ea-9319-ae75fa13b7d3.png)

If I [add the property at the top level](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEBuDGD2A7WBDALpAXGA2qCEkAZirAKYbbCRoCeADmVlIvALYCWiKANpADRQiHMjwAmzYpAC+-cAUgBnRrGb4ChHilpkATszzyNhNil0BrNVDqNJPLkzCyjxyGWTwxXAOZqXxqAQeeH0qfwDrBiZsSFZObj45CNdhUQkY1XCNaSyncNkwdQDIUwsrGijJWA5dWB5HZ2S3Dy9EXzDkwiCQv06FG2iWdi5eAVyNYhFxKshxvOScgOkAXRdFp2WkqF0yRXgeaEHqeu93dMLIbtCoLjEyRkQ7xAxpRqgxdBQ1GF4AV12DNQiJIAIwCQKSACCMkEQMkACZwZdJAAhGTLHI5IA), it doesn't do anything.

**Comment by kanitw** [comment]
This is the limitation of how independent works (that they can't be further merged).  

Since this sounds like an uncommon case, and it would require fundamental change to how guide merging works, I don't think we will fix it. 

That said, if you have a very convincing use case, please comment here and we could revisit if we want to do this in the future.

**Comment by dechamps** [comment]
Well, I originally filed this as a way to fix #6259. However I don't actually need this any more because I managed to find a workarounds for all my use cases thus far (see #6258). I'm worried these workarounds will not generalize well to even more complex charts, but so far I have not come across an unsolvable case. I will report back if I do.
